File: EntitiesCs/Account.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EntitiesCs
{
    public abstract class Account
    {
        private int number;
        protected decimal balance;  // permite acceder a este campo en las subclases
        public Account()
        {
            Enabled = true;
        }
        public Account(int number, decimal balance) : this()    // encadena  el constructor sin parámetros
        {
            Number = number;
            this.balance = balance;
        }
        public int Number
        {
            get
            {
                return number;
            }
            set
            {
                if (value < 10000)   // regla para aceptar el valor
                    throw new ArgumentException("Numero de cuenta debe ser 10000 o superior");
                number = value;
            }
        }
        public Customer Customer { get; set; }
        public decimal Balance { get => balance; }
        public bool Enabled { get; set; }
        public abstract void Withdraw(decimal value); //Extracción
        public void Deposit(decimal value)  // metodo virtual permite ser sobreescrito
        {
            balance += value;
        }
    }
}

File: EntitiesVb/Account.vb
Public MustInherit Class Account
    Private _number As Integer
    Protected _balance As Decimal
    Public Sub New()
        _Enabled = True
    End Sub
    Public Sub New(number As Integer, balance As Decimal)
        MyBase.New()
        Me.Number = number
        _balance = balance
    End Sub
    Public Property Number As Integer
        Get
            Return _number
        End Get
        Set(value As Integer)
            If value < 10000 Then
                Throw New ArgumentException("Numero de cuenta debe ser 10000 o superior")
            End If
            _number = value
        End Set
    End Property
    Public ReadOnly Property Balance As Decimal
        Get
            Return _balance
        End Get
    End Property
    Private _customer As Customer
    Public Property Customer As Customer
        Get
            Return _customer
        End Get
        Set(value As Customer)
            _customer = value
        End Set
    End Property
    Public Property Enabled As Boolean
    Public Sub Deposit(value As Decimal)
        _balance += value
    End Sub

    Public MustOverride Sub Withdraw(value As Decimal)

    Public Overrides Function ToString() As String
        Return "Account " & Number
    End Function
End Class
